src/integration-test/java/com/commercetools/sync/integration/externalsource/products/ProductSyncIT.java

```diff
@@ -582,4 +589,48 @@ public void sync_withSameDraftsWithChangesInBatch_ShouldRetryUpdateBecauseOfConc
         assertThat(errorCallBackMessages).isEmpty();
         assertThat(warningCallBackMessages).isEmpty();
     }
+
+    @Test
+    public void sync_withProductBundle_shouldCreateProductReferencingExistingProduct() {
+        final ProductDraft productDraft = createProductDraftBuilder(PRODUCT_KEY_2_RESOURCE_PATH,
+            ProductType.referenceOfId(productType.getKey()))
+            .taxCategory(null)
+            .state(null)
+            .build();
+
+        // Creating the product reference attribute value
+        final ObjectNode productReference = JsonNodeFactory.instance.objectNode();
+        productReference.put(REFERENCE_TYPE_ID_FIELD, Product.referenceTypeId());
+        productReference.put(REFERENCE_ID_FIELD, product.getKey());
```

Comment:
Why don't you use native support of SDK to create Attributes of type reference instead of using abstract ObjectNode?

```
final AttributeDraft ofReference = AttributeDraft.of("product-reference", Reference.of(Product.referenceTypeId(), product.getKey()));

final AttributeDraft ofSetOfReferences = AttributeDraft.of("product-references", asSet(Reference.of(Product.referenceTypeId(), product.getKey())));

public static <T> Set<T> asSet(final T ... params) {
  return new LinkedHashSet<>(Arrays.asList(params));
}
```

Reply:
0c60e4e